Bunkum.Protocols.Http/Socket/SocketHttpListener.cs

```diff
@@ -75,8 +81,53 @@ public override void StartListening()
         }
     }
 
-    private ListenerContext ReadRequestIntoContext(System.Net.Sockets.Socket client, Stream stream)
+    private ListenerContext ReadRequestIntoContext(System.Net.Sockets.Socket client, Stream rawStream)
     {
+        Stream stream = rawStream;
+        SslStream? sslStream = null;
+        if (this._cert != null)
+        {
+            sslStream = new SslStream(rawStream);
+
+            SslServerAuthenticationOptions authOptions = new()
+            {
+                //TODO: move this into the config
+                EnabledSslProtocols = SslProtocols.Tls12 | SslProtocols.Tls11,
+                ServerCertificate = this._cert,
+                ClientCertificateRequired = false,
+                RemoteCertificateValidationCallback = (_, _, _, _) => true,
+            };
+
+            // On Windows we cant set the cipher suites
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                //TODO: move this into the config somehow
```

Comment:
Probably can have an option similar to SECLEVEL that enables groups of ciphers based on the level set.

Don't think we need fine-tuning past that unless someone requests it